### ISSUE TITLE
[clang-tidy][readability-identifier-naming] forward declaration breaks check

### ISSUE THREAD

**Issue by Yaraslaut** [llvm/llvm-project#64463]
Forward declaration does not emit warning for bad style 
```cpp
struct Foo{int v; int s;}; // invalid case style for struct 'Foo' [readability-identifier-naming]
struct foo{int v; int s;};

class FooC{int v; int s;}; //invalid case style for class 'FooC' [readability-identifier-naming]
class foo_c{int v; int s;};


class FooB; // no warning
class FooB{double g;};

struct FooD; // no warning
struct FooD{int v;int s;};

int main()
{

}
```
Code on: [Godbolt](https://godbolt.org/z/beGbGeT8W)

**Comment by PiotrZSL** [comment]
Looks like minimum example is:

```
class FooB;
class FooB{double g;};

struct FooD;
struct FooD{int v;int s;};
```

**Comment by PiotrZSL** [comment]
https://reviews.llvm.org/D157190

**Comment by sfc-gh-sgiesecke** [comment]
It seems like the fix created a new problem. If you have a forward declaration in a non-system header, and the actual definition in a system header, it wasn't diagnosed before, but it is now.

**Comment by Yaraslaut** [comment]
> It seems like the fix created a new problem. If you have a forward declaration in a non-system header, and the actual definition in a system header, it wasn't diagnosed before, but it is now.

I suppose, that if you can not change declarations in external header files you can disable clang-tidy checks with `NOLINT`

**Comment by sfc-gh-sgiesecke** [comment]
> > It seems like the fix created a new problem. If you have a forward declaration in a non-system header, and the actual definition in a system header, it wasn't diagnosed before, but it is now.
> 
> I suppose, that if you can not change declarations in external header files you can disable clang-tidy checks with `NOLINT`

You can, but this shouldn't be necessary.
